golem-worker-service-base/src/api/api_definition.rs

```diff
@@ -479,7 +458,7 @@ impl From<HttpMiddlewares> for MiddlewareData {
 
         for i in value.0.iter() {
             match i {
-                HttpMiddleware::AddCorsHeaders(cors0) => cors = Some(cors0.clone()),
+                HttpMiddleware::Cors(cors0) => cors = Some(cors0.clone()),
```

Comment:
CORS is not just sending preflight response or  adding response headers to other requests, although most browsers acknowledge it and disallow further requests (after preflight)
So renamed it to CORS, that it has things to do during `middleware_in` and `middleware_out`. The impl is fully inspired from `poem` (further down below)